plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/ElasticsearchClient.java

```diff
@@ -542,7 +542,7 @@ public String executeQuery(String index, String query)
         try {
             response = client.getLowLevelClient()
                     .performRequest(
-                            "GET",
```

Comment:
The use of GET is intentional and is the documented form in the ES API docs: https://www.elastic.co/guide/en/elasticsearch/reference/current/search-search.html#search-search-api-example

Which load balancers cause problems?

Reply:
![image](https://github.com/trinodb/trino/assets/66972/43e33d0d-901d-4454-b3b2-27c9e71de557)

Post is documented as well and more correct if we transmit a request body

Reply:
It's not more correct -- POST is not idempotent, so "technically" the operation can't be retried. There's nothing wrong per with sending a body with GET.

I've never heard of any issues with this since I wrote the plugin a few years ago, so I'm trying to understand how widespread of a problem this is is, or is it just a one-off issue with a buggy load balancer.

Reply:
We can make it configurable in these cases where "buggy" lb is breaking the plugin. Seems pretty simple to implement

Reply:
Google Cloud LB is a notable example of LB that disallows such requests. Search `body_not_allowed` on https://cloud.google.com/load-balancing/docs/https/https-logging-monitoring

Reply:
So... config toggle?

Reply:
nope, default to POST IMO.

Reply:
@martint disagrees

Reply:
That’s too bad. I was hoping it was a niche lb implementation that didn’t accept those types of requests.

A configuration option is not a good solution. Every new config option adds complexity that the user must care about.

I’m ok with switching to POST and giving that a try in light of this. We should check if there’s any retry mechanism that needs to be adjusted, as POST is not an inherently retryable operation.